sdk/core/azure-core-test/src/main/java/com/azure/core/test/http/TestProxyPlaybackClient.java

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.test.http;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpMethod;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.test.utils.HttpURLConnectionHttpClient;
+import com.azure.core.test.utils.TestProxyUtils;
+import com.azure.core.util.Context;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import reactor.core.publisher.Mono;
+
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+import java.util.stream.Collectors;
+
+/**
+ * A {@link HttpClient} that plays back test recordings from the external test proxy.
+ */
+public class TestProxyPlaybackClient implements HttpClient {
+
+    private final HttpURLConnectionHttpClient client = new HttpURLConnectionHttpClient();
+    private String xRecordingId;
+    private static final SerializerAdapter SERIALIZER = new JacksonAdapter();
+
+    /**
+     * Starts playback of a test recording.
+     * @param recordFile The name of the file to read.
+     * @param textReplacementRules Rules for replacing text in the playback.
+     * @return A {@link Queue} representing the variables in the recording.
+     * @throws RuntimeException if an {@link IOException} is thrown.
+     */
+    public Queue<String> startPlayback(String recordFile, Map<String, String> textReplacementRules) {
+        // TODO: replacement rules
+        HttpRequest request = new HttpRequest(HttpMethod.POST, String.format("%s/playback/start", TestProxyUtils.getProxyUrl()))
+            .setBody(String.format("{\"x-recording-file\": \"%s\"}", recordFile));
+        try (HttpResponse response = client.sendSync(request, Context.NONE)) {
+            xRecordingId = response.getHeaderValue("x-recording-id");
+            String body = response.getBodyAsString().block();
+            return SERIALIZER.<Map<String, String>>deserialize(body, Map.class, SerializerEncoding.JSON).entrySet().stream().sorted(Comparator.comparingInt(e -> Integer.parseInt(e.getKey()))).map(Map.Entry::getValue).collect(Collectors.toCollection(LinkedList::new));
```

Comment:
Could you add comments to this logic or break it down into a non-stream set of calls to make it clearer what is going on

Reply:
Added a comment. I also am happy to break it down to non-streams if that's better.

Reply:
I think this would be good to break down into non-streams just so the code is easier to understand and change in the future if needed (aka such as when we remove JacksonAdapter)